tests/e2e/framework/kubernetes.go

```diff
@@ -754,12 +809,6 @@ func (k *KubeInfo) deployIstio() error {
 		}
 	}
 
-	// Create istio-system namespace
```

Comment:
This may break the CNI stuff below?

Reply:
there are two branches.  --use_operator and not.  I have not implemented for the scenario where cni is set and operator is set.  IMO if we  we with to test this scenario, that is something we could tackle as follow-on work.  I will post results of e2e.sh tested against this PR for `make e2e` with the cni repo though.

Cheers
-steve

Reply:
Thats not what I mean, for the CNI test before we create the namespace, then install CNI in the namespace. Now we install cni in the namespace, then create the namespace. This may not work, unless CNI is also creating the namespace?

Reply:
i am running e2e cni now.  Actually I am building a buildtools image because e2e for cni repo is broken as a result of a recent build tools change around the /home dir.  (the cni tests create /home/.helm).  I will report back the results as well as my diff.

Reply:
e2e cni image regression fix: https://github.com/istio/tools/pull/441

Reply:
e2e cni seems to work:
```
make e2e
... output deleted ...
0-16T00:47:50.055552Z	info	Safe to ignore resource not found errors in kubectl delete -f /tmp/tmp.zFwUXlNaw8/simple-auth-test-4371d80f12b54ff5af83/yaml/istio-auth.yaml
2019-10-16T00:47:50.055564Z	info	Deleting namespace istio-system
2019-10-16T00:48:25.331008Z	info	Namespace istio-system deletion status: true
2019-10-16T00:48:25.331061Z	info	Cleanup complete
ok  	istio.io/istio/tests/e2e/tests/simple	296.538s
+ popd
/work
+ cleanup_kind_cluster
+ kind export logs --name istio-testing /tmp/tmp.zFwUXlNaw8/kind
Exported logs to: /tmp/tmp.zFwUXlNaw8/kind
+ kind delete cluster --name=istio-testing
Deleting cluster "istio-testing" ...
$KUBECONFIG is still set to use /home/.kube/kind-config-istio-testing even though that file has been deleted, remember to unset it
sdake@beast-01:~/go/src/istio.io/cni$ 
```

Reply:
my cni diff tested with:
```
sdake@beast-01:~/go/src/istio.io/cni$ git diff
diff --git a/Makefile b/Makefile
index cadc120..6da2153 100644
--- a/Makefile
+++ b/Makefile
@@ -56,7 +56,7 @@ ifeq ($(BUILD_WITH_CONTAINER),1)
 export TARGET_OUT = /work/out/$(TARGET_ARCH)_$(TARGET_OS)
 CONTAINER_CLI ?= docker
 DOCKER_SOCKET_MOUNT ?= -v /var/run/docker.sock:/var/run/docker.sock
-IMG ?= gcr.io/istio-testing/build-tools:2019-10-11T13-37-52
+IMG ?= gcr.io/istio-testing/build-tools:2019-10-15T17-10-27
 UID = $(shell id -u)
 PWD = $(shell pwd)
 
diff --git a/test/e2e.sh b/test/e2e.sh
index 8b43027..ec19dcf 100755
--- a/test/e2e.sh
+++ b/test/e2e.sh
@@ -59,7 +59,8 @@ kubectl get pods --all-namespaces -o wide
 ISTIO_DIR="${GOPATH}/src/istio.io/istio"
 if [[ ! -d "${ISTIO_DIR}" ]]
 then
-  git clone https://github.com/istio/istio.git "${ISTIO_DIR}"
+  git clone https://github.com/sdake/istio.git "${ISTIO_DIR}"
+  git checkout operator_e2e
 fi
 pushd "${ISTIO_DIR}" || exit
   make istioctl
```

Reply:
I just recognized the e2e tests in BUILD_WTIH_CONTAINER in the istio/cni repo use cached results.  I am in the process of a reboot so I may `docker volume prune`.

Reply:
https://github.com/istio/istio/issues/17906

Reply:
Seems like cni tests pass. Makes me question the results of the cni e2e test cases, although I have not investigated in much detail what the local test does.  What would you recommend next?